nix/scripts/setup.sh

```diff
@@ -38,6 +40,33 @@ nix_install() {
         echo "Please see: https://nixos.org/nix/manual/#chap-installation" >&2
         exit 1
     fi
+
+    # Additional fixes
+    nix_create_user_dirs
+    nix_make_user_trusted
+}
+
+# Temporary fix for a known issue with Nix 2.14.
+# https://github.com/status-im/status-mobile/issues/16344
+nix_create_user_dirs() {
+    echo -e "Creating missing user profile and gcroot..." >&2
+    NIX_USER_PROFILE="/nix/var/nix/profiles/per-user/${USER}"
+    NIX_USER_GCROOT="/nix/var/nix/gcroots/per-user/${USER}"
+    sudo mkdir "${NIX_USER_PROFILE}" "${NIX_USER_GCROOT}"
+    sudo chown "${USER}" "${NIX_USER_PROFILE}" "${NIX_USER_GCROOT}"
+}
+
+# Fix for warnings about ignoring binary cache.
+nix_make_user_trusted() {
+    # Singl-user installations do not have this issue.
+    [[ ! -f /etc/nix/nix.conf ]] && return
+    echo -e "Adding ${USER} to trusted-users..." >&2
+    if grep trusted-users /etc/nix/nix.conf 2>/dev/null; then
+        sed -i "s/trusted-users = \(.*\)$/trusted-users = \1 $USER/" /etc/nix/nix.conf
+    else
+        echo "trusted-users = $USER" | sudo tee -a /etc/nix/nix.conf >/dev/null
+    fi
+    sudo systemctl restart nix-daemon
```

Comment:
`systemctl` doesn't work for macOS to restart nix-daemon

```nix
if [[ "$(uname -s)" == "Darwin" ]]; then
        echo "Restarting Nix daemon Launchd service" >&2
        launchctl unload /Library/LaunchDaemons/org.nixos.nix-daemon.plist
        launchctl load   /Library/LaunchDaemons/org.nixos.nix-daemon.plist
elif [[ "$(uname -s)" == "Linux" ]] && [[ "$(nix_install_type)" == "multi" ]]; then
    echo "Restarting Nix daemon Systemd service" >&2
    systemctl daemon-reload
    systemctl restart nix-daemon
fi
```

Reply:
After the `systemctl` fix.
`make run-clojure` fails with 

```log
ok      github.com/git-lfs/git-lfs/v3/lfs       4.831s
--- FAIL: TestDoWithAuthApprove (0.00s)
panic: httptest: failed to listen on a port: listen tcp6 [::1]:0: bind: operation not permitted [recovered]
        panic: httptest: failed to listen on a port: listen tcp6 [::1]:0: bind: operation not permitted

goroutine 8 [running]:
testing.tRunner.func1.2({0x13ea8c0, 0xc000192f60})
        /nix/store/p2q9yykn507lh6hkvm5zl4nz0jxrlvd0-go-1.19.9/share/go/src/testing/testing.go:1396 +0x24e
testing.tRunner.func1()
        /nix/store/p2q9yykn507lh6hkvm5zl4nz0jxrlvd0-go-1.19.9/share/go/src/testing/testing.go:1399 +0x39f
panic({0x13ea8c0, 0xc000192f60})
        /nix/store/p2q9yykn507lh6hkvm5zl4nz0jxrlvd0-go-1.19.9/share/go/src/runtime/panic.go:884 +0x212
net/http/httptest.newLocalListener()
        /nix/store/p2q9yykn507lh6hkvm5zl4nz0jxrlvd0-go-1.19.9/share/go/src/net/http/httptest/server.go:71 +0xe9
net/http/httptest.NewUnstartedServer(...)
        /nix/store/p2q9yykn507lh6hkvm5zl4nz0jxrlvd0-go-1.19.9/share/go/src/net/http/httptest/server.go:119
net/http/httptest.NewServer({0x152cea0?, 0xc000012930})
        /nix/store/p2q9yykn507lh6hkvm5zl4nz0jxrlvd0-go-1.19.9/share/go/src/net/http/httptest/server.go:106 +0x2d
github.com/git-lfs/git-lfs/v3/lfsapi.TestDoWithAuthApprove(0xc0000e7a00)
        /private/tmp/nix-build-git-lfs-3.3.0.drv-0/source/lfsapi/auth_test.go:62 +0xc5
testing.tRunner(0xc0000e7a00, 0x14b4ce8)
        /nix/store/p2q9yykn507lh6hkvm5zl4nz0jxrlvd0-go-1.19.9/share/go/src/testing/testing.go:1446 +0x10b
created by testing.(*T).Run
        /nix/store/p2q9yykn507lh6hkvm5zl4nz0jxrlvd0-go-1.19.9/share/go/src/testing/testing.go:1493 +0x35f
FAIL    github.com/git-lfs/git-lfs/v3/lfsapi    0.671s
FAIL
error: builder for '/nix/store/6pr30wwq3nz6vyvpiqc8if652ps5b44l-git-lfs-3.3.0.drv' failed with exit code 1;
       last 10 log lines:
       > net/http/httptest.NewServer({0x152cea0?, 0xc000012930})
       >         /nix/store/p2q9yykn507lh6hkvm5zl4nz0jxrlvd0-go-1.19.9/share/go/src/net/http/httptest/server.go:106 +0x2d
       > github.com/git-lfs/git-lfs/v3/lfsapi.TestDoWithAuthApprove(0xc0000e7a00)
       >     /private/tmp/nix-build-git-lfs-3.3.0.drv-0/source/lfsapi/auth_test.go:62 +0xc5
       > testing.tRunner(0xc0000e7a00, 0x14b4ce8)
       >       /nix/store/p2q9yykn507lh6hkvm5zl4nz0jxrlvd0-go-1.19.9/share/go/src/testing/testing.go:1446 +0x10b
       > created by testing.(*T).Run
       >         /nix/store/p2q9yykn507lh6hkvm5zl4nz0jxrlvd0-go-1.19.9/share/go/src/testing/testing.go:1493 +0x35f
       > FAIL github.com/git-lfs/git-lfs/v3/lfsapi    0.671s
       > FAIL
       For full logs, run 'nix-store -l /nix/store/6pr30wwq3nz6vyvpiqc8if652ps5b44l-git-lfs-3.3.0.drv'.
error: 1 dependencies of derivation '/nix/store/9h2b6nqyyqhw6182zif98lrvjjgvx6zq-nix-prefetch-git.drv' failed to build
error: 1 dependencies of derivation '/nix/store/h1l12lz8dgb7vjnjqvg67i9fd1r7jz85-yarn2nix-1.0.0.drv' failed to build
error:
       … while calling the 'derivationStrict' builtin

         at //builtin/derivation.nix:9:12: (source not available)

       … while evaluating derivation 'status-mobile-shell'
         whose name attribute is located at /nix/store/vcj90ksw5vk6ia960b73vpncxa9pw826-source/pkgs/stdenv/generic/make-derivation.nix:303:7

       … while evaluating attribute 'shellHook' of derivation 'status-mobile-shell'

         at /Users/siddarthkumar/code/status-mobile/nix/lib/mergeSh.nix:19:5:

           18|     # shellHook is a string, not a list
           19|     shellHook = (super.shellHook or "") + concatStrings (catAttrs "shellHook" shells);
             |     ^
           20|   })

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: 1 dependencies of derivation '/nix/store/psmqz8kw0kpx9wk09sn5fzc955jvd0fh-yarn.nix.drv' failed to build
make: *** [run-clojure] Error 1
```

Reply:
Yeah.

Reply:
You're mixing two issues in one comment.